File: cli-ios/RemoteCoding/DesignSystem/Components.swift
import SwiftUI

/// 字母头像(圆角方块)。
struct Avatar: View {
    let letter: String
    let color: Color
    var size: CGFloat = 44
    var body: some View {
        RoundedRectangle(cornerRadius: size * 0.28)
            .fill(color.gradient)
            .frame(width: size, height: size)
            .overlay(
                Text(letter)
                    .font(.system(size: size * 0.42, weight: .bold))
                    .foregroundStyle(.white)
            )
    }
}

/// 状态徽章(图标 + 文字,半透明底色)。
struct StatusBadge: View {
    let status: TaskStatus
    var body: some View {
        HStack(spacing: 4) {
            Image(systemName: status.icon).font(.system(size: 11, weight: .semibold))
            Text(status.label).font(.system(size: 12, weight: .semibold))
        }
        .foregroundStyle(status.color)
        .padding(.horizontal, 9).padding(.vertical, 5)
        .background(status.color.opacity(0.14))
        .clipShape(Capsule())
    }
}

/// 统计卡(待处理 / 运行中 / 已完成)。
struct StatCard: View {
    let icon: String
    let label: String
    let value: Int
    let color: Color
    var body: some View {
        VStack(alignment: .leading, spacing: 10) {
            HStack(spacing: 6) {
                Image(systemName: icon).font(.system(size: 13)).foregroundStyle(color)
                Text(label).font(.system(size: 13)).foregroundStyle(Theme.textSec)
            }
            Text("\(value)").font(.system(size: 26, weight: .bold)).foregroundStyle(Theme.text)
        }
        .frame(maxWidth: .infinity, alignment: .leading)
        .padding(14)
        .cardStyle()
    }
}

/// 主操作按钮。
struct ActionButton: View {
    let label: String
    let style: ActionStyle
    var action: () -> Void = {}
    var body: some View {
        Button(action: action) {
            Text(label)
                .font(.system(size: 14, weight: .semibold))
                .foregroundStyle(style.fg)
                .padding(.horizontal, 18).padding(.vertical, 9)
                .background(style.bg)
                .clipShape(RoundedRectangle(cornerRadius: 10))
        }
        .buttonStyle(.plain)
    }
}

/// 任务卡(设计图首页列表项)。
struct TaskCard: View {
    let task: TaskItem
    var body: some View {
        VStack(alignment: .leading, spacing: 14) {
            HStack(spacing: 12) {
                Avatar(letter: task.letter, color: task.avatarColor)
                VStack(alignment: .leading, spacing: 2) {
                    Text(task.name).font(.system(size: 16, weight: .semibold)).foregroundStyle(Theme.text)
                    Text(task.agent).font(.system(size: 13)).foregroundStyle(Theme.textSec)
                }
                Spacer()
                StatusBadge(status: task.status)
            }
            HStack {
                Text(task.detail).font(.system(size: 14)).foregroundStyle(Theme.textSec)
                    .lineLimit(1)
                Spacer()
                ActionButton(label: task.actionLabel, style: task.actionStyle)
            }
        }
        .padding(16)
        .cardStyle()
    }
}

/// 区块卡头部(图标 + 标题)。
struct SectionHeader: View {
    let icon: String
    let title: String
    var tint: Color = Theme.textSec
    var body: some View {
        HStack(spacing: 8) {
            Image(systemName: icon).font(.system(size: 14)).foregroundStyle(tint)
            Text(title).font(.system(size: 15, weight: .semibold)).foregroundStyle(tint == Theme.textSec ? Theme.text : tint)
            Spacer()
        }
    }
}

/// 通用区块卡容器。
struct SectionCard<Content: View>: View {
    let icon: String
    let title: String
    var tint: Color = Theme.textSec
    @ViewBuilder var content: Content
    var body: some View {
        VStack(alignment: .leading, spacing: 12) {
            SectionHeader(icon: icon, title: title, tint: tint)
            content
        }
        .padding(16)
        .frame(maxWidth: .infinity, alignment: .leading)
        .cardStyle()
    }
}

/// 通知行。
struct NotificationRow: View {
    let noti: AppNotification
    var body: some View {
        HStack(alignment: .top, spacing: 12) {
            Image(systemName: noti.kind.icon)
                .font(.system(size: 18))
                .foregroundStyle(noti.kind.color)
                .frame(width: 24)
            VStack(alignment: .leading, spacing: 3) {
                Text(noti.title).font(.system(size: 15, weight: .semibold)).foregroundStyle(Theme.text)
                Text(noti.subtitle).font(.system(size: 13)).foregroundStyle(Theme.textSec)
            }
            Spacer()
            VStack(alignment: .trailing, spacing: 6) {
                Text(noti.time).font(.system(size: 12)).foregroundStyle(Theme.textTer)
                if noti.unread {
                    Circle().fill(Theme.blue).frame(width: 7, height: 7)
                }
            }
        }
        .padding(14)
        .cardStyle()
    }
}

File: cli-anr/app/src/main/java/com/aicodingremote/app/designsystem/Components.kt
package com.aicodingremote.app.designsystem

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.MoreHoriz
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Icon
import androidx.compose.material3.OutlinedTextFieldDefaults
import androidx.compose.material3.Text
import androidx.compose.material3.TextFieldColors
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.aicodingremote.app.models.AppNotification
import com.aicodingremote.app.models.TaskStatus

/** 字母头像(圆角方块)。对位 iOS `Avatar`。 */
@Composable
fun Avatar(
    letter: String,
    color: Color,
    size: Dp = 44.dp,
) {
    Box(
        modifier = Modifier
            .size(size)
            .clip(RoundedCornerShape(size * 0.28f))
            .background(Brush.linearGradient(listOf(color, color.copy(alpha = 0.78f)))),
        contentAlignment = Alignment.Center,
    ) {
        Text(
            text = letter,
            color = Color.White,
            fontWeight = FontWeight.Bold,
            fontSize = (size.value * 0.42f).sp,
        )
    }
}

/** 状态徽章(图标 + 文字,半透明底色)。对位 iOS `StatusBadge`。 */
@Composable
fun StatusBadge(status: TaskStatus) {
    Row(
        modifier = Modifier
            .clip(CircleShape)
            .background(status.color.copy(alpha = 0.14f))
            .padding(horizontal = 9.dp, vertical = 5.dp),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(4.dp),
    ) {
        Icon(status.icon, contentDescription = null, tint = status.color, modifier = Modifier.size(11.dp))
        Text(status.label, color = status.color, fontSize = 12.sp, fontWeight = FontWeight.SemiBold)
    }
}

/** 通用区块卡头部(图标 + 标题)。对位 iOS `SectionHeader`。 */
@Composable
fun SectionHeader(
    icon: ImageVector,
    title: String,
    tint: Color = Theme.textSec,
) {
    Row(
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(8.dp),
    ) {
        Icon(icon, contentDescription = null, tint = tint, modifier = Modifier.size(14.dp))
        Text(
            title,
            color = if (tint == Theme.textSec) Theme.text else tint,
            fontSize = 15.sp,
            fontWeight = FontWeight.SemiBold,
        )
        Spacer(Modifier.weight(1f))
    }
}

/** 通用区块卡容器。对位 iOS `SectionCard`。 */
@Composable
fun SectionCard(
    icon: ImageVector,
    title: String,
    tint: Color = Theme.textSec,
    content: @Composable () -> Unit,
) {
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .cardStyle()
            .padding(16.dp),
        verticalArrangement = Arrangement.spacedBy(12.dp),
    ) {
        SectionHeader(icon = icon, title = title, tint = tint)
        content()
    }
}

/** 通知行。对位 iOS `NotificationRow`。 */
@Composable
fun NotificationRow(noti: AppNotification) {
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .cardStyle()
            .padding(14.dp),
        verticalAlignment = Alignment.Top,
        horizontalArrangement = Arrangement.spacedBy(12.dp),
    ) {
        Icon(
            noti.kind.icon,
            contentDescription = null,
            tint = noti.kind.color,
            modifier = Modifier.size(18.dp),
        )
        Column(Modifier.weight(1f), verticalArrangement = Arrangement.spacedBy(3.dp)) {
            Text(noti.title, color = Theme.text, fontSize = 15.sp, fontWeight = FontWeight.SemiBold)
            Text(noti.subtitle, color = Theme.textSec, fontSize = 13.sp)
        }
        Column(horizontalAlignment = Alignment.End, verticalArrangement = Arrangement.spacedBy(6.dp)) {
            Text(noti.time, color = Theme.textTer, fontSize = 12.sp)
            if (noti.unread) {
                Box(Modifier.size(7.dp).clip(CircleShape).background(Theme.blue))
            }
        }
    }
}

/** 中性(描边)按钮 —— 复用组件。对位 iOS `NeutralButton`。 */
@Composable
fun NeutralButton(
    label: String,
    onClick: () -> Unit,
    modifier: Modifier = Modifier,
) {
    Button(
        onClick = onClick,
        modifier = modifier
            .fillMaxWidth()
            .height(46.dp),
        shape = RoundedCornerShape(12.dp),
        colors = ButtonDefaults.buttonColors(
            containerColor = Theme.cardHi,
            contentColor = Theme.text,
        ),
        border = androidx.compose.foundation.BorderStroke(1.dp, Theme.stroke),
    ) {
        Text(label, fontSize = 15.sp, fontWeight = FontWeight.SemiBold)
    }
}

/** 顶部「更多」图标按钮(详情页右上的 ellipsis)。 */
@Composable
fun MoreIcon(tint: Color = Theme.text) {
    Icon(Icons.Default.MoreHoriz, contentDescription = "更多", tint = tint, modifier = Modifier.size(20.dp))
}

/** 暗色输入框配色(供 OutlinedTextField 复用)。 */
@Composable
fun darkFieldColors(): TextFieldColors = OutlinedTextFieldDefaults.colors(
    focusedTextColor = Theme.text,
    unfocusedTextColor = Theme.text,
    focusedBorderColor = Theme.blue,
    unfocusedBorderColor = Theme.stroke,
    focusedContainerColor = Theme.field,
    unfocusedContainerColor = Theme.field,
    cursorColor = Theme.blue,
    focusedPlaceholderColor = Theme.textTer,
    unfocusedPlaceholderColor = Theme.textTer,
)
